packages/xchain-ethereum/src/client.ts

```diff
@@ -102,108 +165,88 @@ export default class Client implements EthereumClient {
    * changes the provider
    */
   EtherscanProvider(): Provider {
-    const newWallet = this.wallet.connect(new EtherscanProvider(this._network))
-    this.changeWallet(newWallet)
-    return (this._provider = this._wallet.provider)
+    this._provider = new EtherscanProvider(this._network)
+    this.changeWallet(this.wallet.connect(this._provider))
+    return this._provider
   }
 
   /**
    * Set's the current network
    */
-  setNetwork(network: Network): Network {
+  setNetwork(network: XChainNetwork): void {
     if (!network) {
       throw new Error('Wallet must be provided')
     } else {
-      this._network = network
-      this._provider = getDefaultProvider(network)
-      this._etherscan = new EtherscanProvider(network)
-      return this._network
+      this._network = xchainNetworkToEths(network)
+      this._provider = getDefaultProvider(this._network)
+      this._etherscan = new EtherscanProvider(this._network)
     }
   }
 
   /**
    * Set's the current vault contract
    */
-  setVault(vault: string): ethers.Contract {
+  setVault(vault: string): string {
     if (!vault) {
       throw new Error('Vault address must be provided')
-    } else {
-      const contract = new ethers.Contract(vault, vaultABI, this._provider)
-      this._vault = contract.connect(this.wallet)
-      return this._vault
     }
-  }
-
-  /**
-   * Generates a new mnemonic / phrase
-   */
-  static generatePhrase(): Phrase {
-    return generateMnemonic()
-  }
 
-  /**
-   * Validates a mnemonic phrase
-   */
-  static validatePhrase(phrase: Phrase): boolean {
-    return validateMnemonic(phrase) ? true : false
+    return (this._vault = vault)
   }
 
   /**
    * Sets a new phrase (Eg. If user wants to change wallet)
    */
-  setPhrase(phrase: Phrase): boolean {
-    if (!Client.validatePhrase(phrase)) {
+  setPhrase(phrase: string): Address {
+    if (!Crypto.validatePhrase(phrase)) {
       throw new Error('Phrase must be provided')
-    } else {
-      this._phrase = phrase
-      const newWallet = ethers.Wallet.fromMnemonic(phrase)
-      this.changeWallet(newWallet)
-      return true
     }
+
+    this._phrase = phrase
+    this.changeWallet(ethers.Wallet.fromMnemonic(phrase))
+    return this.getAddress()
   }
 
   /**
    * Validates an address
    */
   static validateAddress(address: Address): boolean {
```

Comment:
Make `validateAddress` a public (not `static`) method (to be in consistent with other clients).